pkg/sql/create_table.go

```diff
@@ -1078,6 +1083,28 @@ func ResolveFK(
 	return nil
 }
 
+// findTargetConstraintColumnIDs returns columnIDs as they are defined on the target table.
+// It is important to use them as defined on that table, so we use them in the order they are defined.
+// Otherwise, references created with columns defined out of order will fail foreign key checks.
+func findTargetConstraintColumnIDs(targetUniqueConstraint descpb.UniqueConstraint, target *tabledesc.Mutable) []ColumnID {
```

Comment:
Further testing raised this point to me. In order to use a referenced index, it's necessary that the columnIDs on the FK definition are in the same order as on the target table. FK Checks fail otherwise. Rather than changing the FK checks in each place they were used, it seemed easier to change them here, once, at the point of definition.

I'm not very keen on this implementation, but I don't see a better way. Perhaps you have some ideas about other ways that we can get the data we need?
We could extend `descpb.UniqueConstraint` for instance, to include a method that returns the target constraint definition?

Reply:
i'm not opposed to making the constraints return the definition if you think it makes it cleaner.

Reply:
`descpb.UniqueConstraint` now returns ColumnIDs. This did require adding the method to all found implementations, but they all had the ColumnIDs field so each is a very simple getter.